conans/model/build_info.py

```diff
@@ -363,6 +363,19 @@ def set_relative_base_folder(self, folder):
                             updates[prop_name] = os.path.join(folder, value)
                 component._generator_properties.update(updates)
 
+    def deploy_base_folder(self, package_folder, deploy_folder):
+        """Prepend the folder to all the directories"""
+        for component in self.components.values():
+            for varname in _DIRS_VAR_NAMES:
+                origin = getattr(component, varname)
+                if origin is not None:
+                    new_ = []
+                    for el in origin:
+                        rel_path = os.path.relpath(el, package_folder)
+                        new_.append(os.path.join(deploy_folder, rel_path))
+                    origin[:] = new_
+                # TODO: Missing properties
```

Comment:
I think we should refactor this method and the `set_relative_base_folder()` one in the future. They're very similar and we could have a single method to set a general base folder.

Reply:
agree